### ISSUE TITLE
[question] Support for selection of pinned rows

### ISSUE THREAD

**Issue by andreac92** [mui/mui-x#10057]
### Order ID or Support key 💳 (optional)

50967

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

According to the docs, pinned rows currently do not support row selection: https://mui.com/x/react-data-grid/row-pinning/#usage-with-other-features

Our question is -- are there any future plans to support this, or, is there a way we can work around this? We're working on a feature where we want to be able to checkbox select pinned rows and take actions on them. But as of now the pinned rows don't render the checkboxes.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: macOS 13.2.1
  Binaries:
    Node: 16.19.0 - ~/.nvm/versions/node/v16.19.0/bin/node
    Yarn: Not Found
    npm: 8.19.3 - ~/.nvm/versions/node/v16.19.0/bin/npm
  Browsers:
    Chrome: 115.0.5790.170
    Edge: Not Found
    Safari: 16.3
  npmPackages:
    @emotion/react: ^11.10.4 => 11.11.0 
    @emotion/styled: ^11.10.4 => 11.11.0 
    @mui/base:  5.0.0-beta.1 
    @mui/core-downloads-tracker:  5.13.1 
    @mui/material: ^5.10.5 => 5.13.1 
    @mui/private-theming:  5.13.1 
    @mui/styled-engine:  5.12.3 
    @mui/system:  5.13.1 
    @mui/types:  7.2.4 
    @mui/utils:  5.14.1 
    @mui/x-data-grid:  6.10.1 
    @mui/x-data-grid-pro: ^6.7.0 => 6.10.1 
    @mui/x-date-pickers:  6.5.0 
    @mui/x-date-pickers-pro: ^6.4.0 => 6.5.0 
    @mui/x-license-pro: ^6.6.0 => 6.10.0 
    @types/react: ^18.0.0 => 18.2.6 
    react: ^18.0.0 => 18.2.0 
    react-dom: ^18.0.0 => 18.2.0 
    typescript: ^4.6.3 => 4.9.5 
```
</details>

**Comment by romgrk** [comment]
@mui/xgrid Does someone have more context on this?

**Comment by andreac92** [comment]
Any updates here? :)

**Comment by cherniavskii** [comment]
Pinned row selection was extracted out of the Row pinning effort in https://github.com/mui/mui-x/pull/4863 to reduce the scope.
I'm converting this issue to a feature request to check if there's a significant demand for this feature 👍🏻

**Comment by britanyponvelle** [comment]
Any word on if this can be picked up @mui/xgrid ? This would be a huge help.

**Comment by andrewteeoh** [comment]
Pinned row selection would be awesome! We use row selection, but currently have to work around this by pulling the pinned rows out of the table and building selection ourselves in order to perform the same actions we have for non-pinned rows.

**Comment by rivamadan** [comment]
Performing actions on pinned rows seems common enough that it should be expected functionality. It would be very useful for my use case.

**Comment by dshapiro-david** [comment]
+1 for selection of pinned rows :)

**Comment by naxis18** [comment]
Waiting for this functionality also!

**Comment by dshapiro-david** [comment]
[cherniavskii](https://github.com/cherniavskii) the requests are piling in haha 😉

**Comment by DanailH** [comment]
@dshapiro-david we are trying to cover as much as we can. If we are not prioritising this feature now it means that there are more pressing things that we need to finish first. In the end, we have limited resources that we need to deploy in the best way we can, hope you can understand.

**Comment by dshapiro-david** [comment]
@DanailH Absolutely, that makes sense! Ya'll are doing great work. I was just addressing the comment "I'm converting this issue to a feature request to check if there's a significant demand for this feature"

**Comment by edward-hunt** [comment]
+1 for this feature

**Comment by akhalizev** [comment]
And +1 from me. Our team is thinking about using pinning feature, would be nice to have. 
Or if anyone knows workaround they can share?

**Comment by PedroScheid1** [comment]
+1

**Comment by atillaaliyev** [comment]
+1 for this feature

**Comment by gp919934** [comment]
Hello @cherniavskii @oliviertassinari @joserodolfofreitas .Is it possible to Use pinned rows feature in Community version of data-grid?

**Comment by clintonmedbery** [comment]
> [@dshapiro-david](https://github.com/dshapiro-david) we are trying to cover as much as we can. If we are not prioritising this feature now it means that there are more pressing things that we need to finish first. In the end, we have limited resources that we need to deploy in the best way we can, hope you can understand.

<img src="https://static.klipy.com/ii/d7aec6f6f171607374b2065c836f92f4/37/2c/wQtXeIQp.gif"/>
